src/main/java/org/embulk/input/marketo/rest/MarketoRestClient.java

```diff
@@ -435,4 +445,76 @@ private InputStream getBulkExtractResult(MarketoRESTEndpoint endpoint, String ex
         }
         return getRecordWithOffsetPagination(endPoint + MarketoRESTEndpoint.GET_PROGRAMS.getEndpoint(), multimap, ObjectNode.class);
     }
+
+    public List<MarketoField> describeCustomObject(String apiName)
+    {
+        MarketoResponse<ObjectNode> jsonResponse = doGet(endPoint + MarketoRESTEndpoint.GET_CUSTOM_OBJECT_DESCRIBE.getEndpoint(new ImmutableMap.Builder().put("api_name", apiName).build()), null, null, new MarketoResponseJetty92EntityReader<ObjectNode>(this.readTimeoutMillis));
+        if (jsonResponse.getResult().size() == 0) {
+            throw new ConfigException(String.format("Custom Object %s is not exits.", apiName));
+        }
+        List<MarketoField> marketoFields = new ArrayList<>();
+        JsonNode fieldNodes = jsonResponse.getResult().get(0).path("fields");
+        for (JsonNode node : fieldNodes) {
+            String dataType = node.get("dataType").asText();
+            String name = node.get("name").asText();
+            marketoFields.add(new MarketoField(name, dataType));
+        }
+        if (marketoFields.size() == 0) {
+            throw new ConfigException(String.format("Custom Object %s don't have any field data.", apiName));
+        }
+        return marketoFields;
+    }
+    private <T> RecordPagingIterable<T> getRecordWithPagination(final String endPoint, final String customObjectFilterType, final String customObjectFields, final Integer fromValue, final Integer toValue, final Class<T> recordClass)
```

Comment:
Any reason the method `getRecordWithOffsetPagination` is unusable so that you create a similar method?

Reply:
in my opinion, the method `getRecordWithOffsetPagination` is well tested and support the same batch size which is default 300 for all Marketo endpoint